common/src/main/scala/skinny/ParamType.scala

```diff
@@ -79,22 +79,24 @@ object ParamType {
     case v: Short => scala.math.BigDecimal(v)
   })
   case object String extends AbstractParamType({
+    case v: String if v.isEmpty => null // don't trim
     case v: String => v
     case v => v.toString
   })
   case object Byte extends AbstractParamType({
-    case v: String if v == null || v.trim.isEmpty => null
+    case v: String if v.trim.isEmpty => null
     case v: String => v.toByte
     case v: Byte => v
   })
   case object ByteArray extends AbstractParamType({
-    case v: String if v == null || v.trim.isEmpty => null
+    case v: String if v.trim.isEmpty => null
     case v: String => v.getBytes
     case v: Array[Byte] => v
   })
   case object DateTime extends ParamType {
     def unapply(v: Any): Option[Any] = v match {
       case null => None
+      case v: String if v.trim.isEmpty => Some(null)
```

Comment:
http://www.scala-lang.org/api/current/index.html#scala.PartialFunction$

We found that `PartialFunction.condOpt` unexpectedly (for me) returns `Some(null)` in other `ParamType`s. Though this is another issue, let's fix to return `None` in this pull request.